grouper/models.py

```diff
@@ -1274,6 +1367,92 @@ class AsyncNotification(Model):
     send_after = Column(DateTime, nullable=False)
     sent = Column(Boolean, default=False, nullable=False)
 
+    @staticmethod
+    def _get_unsent_expirations(session, now_ts):
+        """Get upcoming group membership expiration notifications as a list of (group_name,
+        member_name, email address) tuples.
+        """
+        tuples = []
+        emails = session.query(AsyncNotification).filter(
+            AsyncNotification.key.like("EXPIRATION%"),
+            AsyncNotification.sent == False,
+            AsyncNotification.send_after < now_ts,
+        ).all()
+        for email in emails:
+            _, group_name, member_name = email.key.split(ILLEGAL_NAME_CHARACTER)
```

Comment:
nit: Might be best to write this as a function and put it next to where you generate the `_expiration_key` for clarity.